cmd/collector/main.go

```diff
@@ -199,8 +199,8 @@ func ensureIdentity(configPath string) error {
 		return fmt.Errorf("unable to interpret config file: %w", err)
 	}
 
-	// if they already have an AgentID, don't generate it
-	if candidateConfig.AgentID != "" {
+	// If the AgentID is not a ULID (legacy ID or empty) then we need to generate a ULID as the AgentID.
+	if _, err := ulid.Parse(candidateConfig.AgentID); err == nil {
 		return nil
 	}
```

Comment:
Kind of a ridiculous question here, but have we considered marshalling the (binary) UUID into a ULID? I know it's not technically correct, but it would create a one-to-one mapping of the legacy ID to the new ULID, which bindplane-op could use to correlate during the update here.

Reply:
@BinaryFissionGames talked with BPOP team on this and they are ok with this creating a new agent. Since it'll likely only be a release or two it didn't seem worth having special code in both BPOP and the collector to mitigate this.